### DISCUSSION TITLE
Subscribing Events Before Request and Handling Event on Receiving Success Response

### DISCUSSION THREAD

**Discussion by sameerjoshi-git** [artilleryio/artillery#2793] [Artillery]
Need to achieve Subscribing of Events (event which will eventually generate after POST request) with beforeRequest and Handling (Need to verify content generated as event take place and captured, not API response) of Event on afterResponse (if API response is Success).
For subscribing and handling of the event using another JS module which will be invoked from processor using object of the module.
For every unique request there will be different subscription and verification of corresponding content is required.
I am able to subscribe to the notification but lost the context as soon as control came out of processor.

beforeRequest: subscribeToEvent();
POST
afterResponse: handleEvent() 

How we can access event subscribed only in beforeRequest but will be available in afterResponse.

**Comment by bernardobridge** [comment]
Hello @sameerjoshi-git 👋 

You should be able to save objects in `context.vars` using custom code, and use it in later parts of the request: https://www.artillery.io/docs/reference/engines/http#loading-custom-code

Without sharing your test script YAML and custom JS code, it's a bit hard to understand what problem you might be running into.

**Reply by sameerjoshi-git** [reply]
Thanks @bernardobridge for the response.
I am not sure If I understand and doing it correctly but do you mean following should work

YAML

config:
  # This is a test server run by team Artillery
  # It's designed to be highly scalable
  target: <EP1>
  phases:
    - duration: 1
      arrivalRate: 1
      rampTo: 1
      name: Warm up phase
  variables:
  processor: "./processor.js"
scenarios:
  - flow:
      - post:
          url: '/token'
          beforeRequest: "getDataAndToken"
          json:        
          capture:              
      - post:
          url: '<EP2>'
          beforeRequest: "checkToken"
          headers:
          afterResponse: "checkResponse"
       
Custom JS

let { utils } = require('./util');

module.exports = {
    getDataAndToken: getDataAndToken,
    checkToken: checkToken,
    checkResponse: checkResponse,
  };


async function  getDataAndToken(requestParams, context,__) {
    //get token
  }

async function checkToken(requestParams, context, __) {
    
	temp = await util.Instance();
	await temp.subscribeToEvent();
	context.vars["temp"] = temp;
	console.log(` LENGTH ==> ${util.event.length}`); //Expected 0
    
    //return next();
  }

  async function checkResponse(requestParams , response,context, __) {
    console.log(` LENGTH ==> ${context.vars["temp"].event.length}`); //Expected non zero
}

**Reply by sameerjoshi-git** [reply]
@bernardobridge @hassy  Can you please help, just for similar/simpler example can we have redis connection built initially and use it through out the run.
